portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -124,4 +125,34 @@ private int getMaxComments(HttpServletRequest request) {
     System.out.println(maxComments);
     return maxComments;
   }
+
+  @Override
+  public void doDelete(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    System.out.println("Receive request");
+    Query query = new Query("Comment");
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    // Loop through all entities and delete them using the key
+    for (Entity entity: results.asIterable()) {
+      Key key = entity.getKey();
+      datastore.delete(key);
+    }        
+
+    // Redirect back to the HTML page.
+    response.sendRedirect("/index.html");
+  }
+
+  @Override
+  public void doPut(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      System.out.println("Receive PUT request");
+      int maxComment = getMaxComments(request);
+      if (maxComment == -1) {
```

Comment:
You may want to check that `maxComment` is less than 0 here, not just equal to -1.

Reply:
Oh the helper function `getMaxComments` will return -1 if the input is invalid, such as a non-numeric input or an integer input less than 1. This is the reason why I check whether the maxComment equals to -1 because -1 represents the error status.

Reply:
Oops, I missed that, thanks.